device/device.yaml

```diff
@@ -4,51 +4,21 @@ components:
       x-include: 
       - '../common/common.yaml#/components/schemas/Named.Object'
       description: >-
-        A container for emulated protocol devices.
+        A container for emulated interfaces and protocol devices.
       type: object
-      required: [container_name]
+      required: [container_name, ethernet]
       properties:
         container_name: 
           description: >-
             The unique name of a Port or Lag object that will contain the 
-            emulated interfaces and/or devices.
+            emulated interfaces and/or protocol devices.
           type: string
           x-constraint:
           - '/components/schemas/Port/properties/name'
           - '/components/schemas/Lag/properties/name'
-        device_count:
-          description: >-
-            The number of emulated protocol devices or interfaces per port.
-            
-            For example if the device_count is 10 and the choice property value is ethernet 
-            then an implementation MUST create 10 ethernet interfaces.
-            The ethernet property is a container for src, dst and eth_type 
-            properties with each on of those properties being a pattern container 
-            for 10 possible values.  
-
-            If an implementation is unable to support the maximum device_count
-            it MUST indicate what the maximum device_count is using the 
-            /results/capabilities API.
-
-            The device_count is also used by the individual child properties 
-            that are a container for a /components/schemas/Device.Pattern.
-          type: integer
-          default: 1
-          minimum: 1
-          maximum: 100000000
         ethernet:
```

Comment:
For some dual stack protocols , we might need a concept of an Interface which comprises of Ethernet , IPv4 and IPv6. However, we seem to have a reference context here so this might be possible also by adding IPv4 and IPv6 reference to the dual stack router which is directly per device ( e.g. for dual-stack LDP router )